hazelcast/src/main/java/com/hazelcast/util/FutureUtil.java

```diff
@@ -276,6 +278,33 @@ public void handleException(Throwable throwable) {
     }
 
     @PrivateApi
+    public static void waitForever(Collection<Future> futuresToWaitFor, ExceptionHandler exceptionHandler) {
+        Collection<Future> futures = new ArrayList<Future>(futuresToWaitFor);
+        while (true) {
+            Iterator<Future> it = futures.iterator();
+            while (it.hasNext()) {
+                Future future = it.next();
+                try {
+                    future.get();
+                } catch (Exception e) {
```

Comment:
I also see counter examples in `FutureUtil`, some inconsistency exists between methods. 
To me your approach makes more sense. 👍